src/TwigComponent/src/Twig/PropsNode.php

```diff
@@ -66,5 +75,25 @@ public function compile(Compiler $compiler): void
             ->write('}')
             ->write('}')
         ;
+
+        // overwrite the context value if a props with a similar name and a default value exist
+        if ($this->hasNode($name)) {
```

Comment:
What is the goals / impact of this ? Would you have an example ?

Reply:
(would that not erase "app" for instance ? maybe already what's happing with class props i'm not sure there)

Reply:
Symfony\UX\TwigComponent\Tests\Integration\ComponentExtensionTest::testComponentPropsOverwriteContextValue

This test illustrates what we solve here. The idea is when you have conflict between a variable from outside the component context and from inside we choose the one from the component context.

Reply:
Thanks! And sorry for the questions, i've been on many different things at once in my head haha